golang/lib.go

```diff
@@ -0,0 +1,306 @@
+package usearch
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+	"unsafe"
+)
+
+/*
+#include <stdlib.h>
+#include "golang.h"
+// avoid some verbosity by using void* index here
+// proper typed pointer in the cpp file definitions
+void* new_index(char* metric_str, int metric_len,
+    char* accuracy_str, int accuracy_len,
+    int dimensions, int capacity, int connectivity,
+    int expansion_add, int expansion_search
+);
+void destroy(void* index);
+
+const char* save(void* index, char* path);
+const char* load(void* index, char* path);
+const char* view(void* index, char* path);
+
+//todo:: q:: is int<->size_t conversion portable?
+int size(void* index);
+int connectivity(void* index);
+int dimensions(void* index);
+int capacity(void* index);
+
+const char* set_capacity(void* index, int capacity);
+const char* add(void* index, int label, float* vec);
+SearchResults search(void* index, float* query, int query_len, int limit);
+
+//#cgo LDFLAGS: -lomp
+//#cgo CPPFLAGS: -fopenmp
+#cgo CPPFLAGS: -I../fp16/include/
+#cgo CPPFLAGS: -I../simsimd/include/
+#cgo CPPFLAGS: -I../include/
+*/
+import "C"
+
+type DistMetric int8
+
+const (
+	L2_SQ DistMetric = iota
+	IP
+	COS
+	Haversine
+)
+
+func (m DistMetric) String() string {
+	switch m {
+	case L2_SQ:
+		return "l2_sq"
+	case IP:
+		return "ip"
+	case COS:
+		return "cos"
+	case Haversine:
+		return "haversine"
+	default:
+		panic("unknown metric")
```

Comment:
We can start with `panic`, but returning `error` is better. Check out the [UStore bindings](https://github.com/unum-cloud/go-ustore/blob/main/internal/ukv.go) for inspiration.